### PR TITLE
New tests.

### PR DESCRIPTION
**Pull request by JakubWlochTuring**

### CODE REVIEW

File: tests/new_tests.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import unittest
+from utils import add, is_even, reverse_string
+
+class TestUtils(unittest.TestCase):
+
```
Comment (JakubWlochTuring) [inline tests/new_tests.py:5]: Add more edge cases for the add() function, such as testing with large numbers, floating-point values, or potential overflow scenarios.

File: tests/new_tests.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        self.assertEqual(add(-1, 1), 0)
+        self.assertEqual(add(0, 0), 0)
+
+    def test_is_even(self):
+        self.assertTrue(is_even(2))
+        self.assertTrue(is_even(100))
+        self.assertFalse(is_even(3))
+        self.assertFalse(is_even(101))
+
+    def test_reverse_string(self):
```
Comment (JakubWlochTuring) [inline tests/new_tests.py:17]: Test reverse_string() with unicode characters, strings with spaces, and very long strings to improve test coverage.

File: tests/new_tests.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        self.assertEqual(reverse_string("hello"), "olleh")
+        self.assertEqual(reverse_string("Python"), "nohtyP")
+        self.assertEqual(reverse_string(""), "")
```
Comment (JakubWlochTuring) [inline tests/new_tests.py:20]: Consider using unittest.main(verbosity=2) to get more detailed test output, or use pytest for more advanced test reporting.

File: tests/new_tests.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+from utils import add, is_even, reverse_string
+
+class TestUtils(unittest.TestCase):
+
+    def test_add(self):
+        self.assertEqual(add(2, 3), 5)
+        self.assertEqual(add(-1, 1), 0)
+        self.assertEqual(add(0, 0), 0)
+
+    def test_is_even(self):
```
Comment (JakubWlochTuring) [inline tests/new_tests.py:11]: Add tests for boundary conditions like testing is_even() with zero and very large even/odd numbers.

File: tests/new_tests.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        self.assertTrue(is_even(2))
+        self.assertTrue(is_even(100))
+        self.assertFalse(is_even(3))
+        self.assertFalse(is_even(101))
+
```
Comment (JakubWlochTuring) [inline tests/new_tests.py:16]: Good test for reverse_string(), but consider adding tests for strings with special characters, Unicode, or multi-byte characters.
